tests/mir-opt/copy-prop/copy_propagation_arg.rs

```diff
@@ -9,25 +8,43 @@ fn dummy(x: u8) -> u8 {
 
 // EMIT_MIR copy_propagation_arg.foo.CopyProp.diff
 fn foo(mut x: u8) {
+    // CHECK-LABEL: fn foo(
+    // CHECK: debug x => [[x:_.*]];
+    // CHECK: [[three:_.*]] = copy [[x]];
+    // CHECK: [[two:_.*]] = dummy(move [[three]])
+    // CHECK: [[x]] = move [[two]];
     // calling `dummy` to make a use of `x` that copyprop cannot eliminate
     x = dummy(x); // this will assign a local to `x`
 }
 
 // EMIT_MIR copy_propagation_arg.bar.CopyProp.diff
 fn bar(mut x: u8) {
+    // CHECK-LABEL: fn bar(
+    // CHECK: debug x => [[x:_.*]];
+    // CHECK: [[three:_.*]] = copy [[x]];
+    // CHECK: dummy(move [[three]])
+    // CHECK: [[x]] = const 5_u8;
     dummy(x);
     x = 5;
 }
 
 // EMIT_MIR copy_propagation_arg.baz.CopyProp.diff
 fn baz(mut x: i32) -> i32 {
+    // CHECK-LABEL: fn baz(
```

Comment:
I'm wondering if we don't need this function for CopyProp.
I guess this test file originally from dest-prop, and in dest-prop the self-assignment is eliminated. But in copy-prop, it is not eliminated.

Reply:
> and adding a CHECK: [[x]] = copy [[x]];?

There is no self-copy instruction in the mir file, but it has `_2 = copy _1` and `_0 = copy _1` (I already added).
Do you mean that I should add `CHECK: _2 = copy [[x]];`?

Reply:
I modified it in https://github.com/rust-lang/rust/pull/135099/commits/7a641377ae492f0cb2dd0be53a397381e7abfb92. Please check if the change matches your thought 🙏